### PR TITLE
Modify tarball scripts to work with RCA

### PR DESCRIPTION
**Pull request by sidheart**
Does this PR include tests? No, these are bash build changes for performance_analyzer

This project is developed under a test-driven workflow, so please refrain from submitting patches without test coverage. If you are not familiar with testing in Python, please raise an issue instead.

### CODE REVIEW

File: elasticsearch/linux_distributions/opendistro-tar-install.sh [hunk 1]
```diff
@@ -21,6 +21,15 @@ bash $ES_HOME/plugins/opendistro_security/tools/install_demo_configuration.sh -y
...
 chmod -R 755 /dev/shm
+chmod a+rw /tmp
+chown -R elasticsearch $ES_HOME/performance-analyzer-rca
+chown elasticsearch $ES_HOME/data/rca_enabled.conf
+mkdir -p "/var/lib/elasticsearch"
+echo 'true' > /var/lib/elasticsearch/performance_analyzer_enabled.conf
+echo 'true' > /var/lib/elasticsearch/rca_enabled.conf
+chown elasticsearch /var/lib/elasticsearch/performance_analyzer_enabled.conf
+chown elasticsearch /var/lib/elasticsearch/rca_enabled.conf
+chown -R elasticsearch "$ES_HOME/performance-analyzer-rca"
```
Comment (peterzhuamazon) [inline elasticsearch/linux_distributions/opendistro-tar-install.sh:32]: This script will not be run by root. Assuming user does not have root permission for these actions.

File: elasticsearch/linux_distributions/opendistro-tar-install.sh [hunk 2]
```diff
@@ -21,6 +21,15 @@ bash $ES_HOME/plugins/opendistro_security/tools/install_demo_configuration.sh -y
 ##Perf Plugin
 chmod 755 $ES_HOME/plugins/opendistro_performance_analyzer/pa_bin/performance-analyzer-agent
 chmod -R 755 /dev/shm
+chmod a+rw /tmp
+chown -R elasticsearch $ES_HOME/performance-analyzer-rca
+chown elasticsearch $ES_HOME/data/rca_enabled.conf
```
Comment (weicongs-amazon) [inline elasticsearch/linux_distributions/opendistro-tar-install.sh:26]: what's the reason for this change? for tarball distribution, any uid can be used for service startup. And the elasticsearch doesn't exist probably.

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 1 above)
Comment (weicongs-amazon) [inline elasticsearch/linux_distributions/opendistro-tar-install.sh:32]: same as above

File: elasticsearch/linux_distributions/opendistro-tar-install.sh [hunk 3]
```diff
@@ -21,6 +21,15 @@ bash $ES_HOME/plugins/opendistro_security/tools/install_demo_configuration.sh -y
...
+mkdir -p "/var/lib/elasticsearch"
+echo 'true' > /var/lib/elasticsearch/performance_analyzer_enabled.conf
+echo 'true' > /var/lib/elasticsearch/rca_enabled.conf
```
Comment (weicongs-amazon) [inline elasticsearch/linux_distributions/opendistro-tar-install.sh:29]: I think it's better to use a relative file path. could we put these configuration into $ES_HOME/performance-analyzer-rca?

Comment (weicongs-amazon) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh:29]: btw, I don't see this change in docker image. is it required?

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 1 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh]: Removed the majority of these, moved to build.sh, only chmod on files the script creates or moves

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 2 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh]: changed to chmod with 755

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 1 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh]: same solution as above

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 3 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh]: removed these, can be addressed in user documentation

File: elasticsearch/linux_distributions/opendistro-tar-build.sh [hunk 4]
```diff
@@ -60,7 +60,20 @@ done
 echo "List available plugins"
 ls -lrt $basedir
 
-# Download Knn lib 
+# Move RCA files
+## Move RCA folder
+cp -r $PACKAGE_NAME-$OD_VERSION/plugins/opendistro_performance_analyzer/performance-analyzer-rca $PACKAGE_NAME-$OD_VERSION
+chmod -R 755 elasticsearch ${PACKAGE_NAME}-${OD_VERSION}/performance-analyzer-rca
```
Comment (weicongs-amazon) [inline elasticsearch/linux_distributions/opendistro-tar-build.sh:66]: remove "elasticsearch"

File: elasticsearch/linux_distributions/opendistro-tar-build.sh [hunk 5]
```diff
@@ -60,7 +60,20 @@ done
...
+## Move agent script directly into ES_HOME/bin
+mv $PACKAGE_NAME-$OD_VERSION/bin/opendistro_performance_analyzer/performance-analyzer-agent-cli $PACKAGE_NAME-$OD_VERSION/bin
+rm -rf $PACKAGE_NAME-$OD_VERSION/bin/opendistro_performance_analyzer
+## Create the rca_enabled.conf file
+mkdir -p ${PACKAGE_NAME}-${OD_VERSION}/data
+touch ${PACKAGE_NAME}-${OD_VERSION}/data/rca_enabled.conf
```
Comment (weicongs-amazon) [inline elasticsearch/linux_distributions/opendistro-tar-build.sh:72]: could we use different folder for this, such as:/ES_HOME/\<rca\>/config/?

Comment (weicongs-amazon) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh:72]: btw, why do we need this file generated in the building process? could we have it in the artifact? what are the current files included in $PACKAGE_NAME-$OD_VERSION/plugins/opendistro_performance_analyzer/performance-analyzer-rca

File: elasticsearch/linux_distributions/opendistro-tar-build.sh (hunk 4 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh]: Done

File: elasticsearch/linux_distributions/opendistro-tar-install.sh [hunk 6]
```diff
@@ -21,6 +21,7 @@ bash $ES_HOME/plugins/opendistro_security/tools/install_demo_configuration.sh -y
 ##Perf Plugin
 chmod 755 $ES_HOME/plugins/opendistro_performance_analyzer/pa_bin/performance-analyzer-agent
 chmod -R 755 /dev/shm
+chmod +x $ES_HOME/bin/performance-analyzer-agent-cli
```
Comment (peterzhuamazon) [inline elasticsearch/linux_distributions/opendistro-tar-install.sh:24]: Better use 755 than +x to all

File: elasticsearch/linux_distributions/opendistro-tar-build.sh (hunk 5 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh:72]: we can't use a different folder, the code currently expects them in these exact locations. we should be able to address this in the next release.

the file is generated in the build process and contains jars that we need to keep separate so that we don't run into classpath issues

File: elasticsearch/linux_distributions/opendistro-tar-install.sh (hunk 6 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-install.sh]: Done

File: elasticsearch/linux_distributions/opendistro-tar-build.sh (hunk 5 above)
Comment (weicongs-amazon) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh:72]: based on the comment from @ktkrg in the pr https://github.com/opendistro-for-elasticsearch/opendistro-build/pull/288, this rca_enabled.conf isn't required, could we remove it?

Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh]: I ran into issues with this when testing a standalone install.

I've removed the .conf file, but kept the logic to create the data folder. I'll test this with our artifacts and this branch.

File: elasticsearch/linux_distributions/opendistro-tar-build.sh [hunk 7]
```diff
@@ -60,7 +60,18 @@ done
 echo "List available plugins"
 ls -lrt $basedir
 
-# Download Knn lib 
+# Move RCA files
```
Comment (peterzhuamazon) [inline elasticsearch/linux_distributions/opendistro-tar-build.sh:63]: Just realize is this an extra comment or you have more logic under this?

File: elasticsearch/linux_distributions/opendistro-tar-build.sh (hunk 5 above)
Comment (weicongs-amazon) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh]: Thanks.

File: elasticsearch/linux_distributions/opendistro-tar-build.sh (hunk 7 above)
Comment (sidheart) [reply elasticsearch/linux_distributions/opendistro-tar-build.sh]: It was meant to be a comment for the whole block. This is unnecessary so I just removed it.

**Comment by weicongs-amazon** [comment]
@sidheart if you see any issue during the testing, feel free to open a new PR
